Docs: Support dark mode loading screen depending on user settings

BieleckiLtd opened the pull request:
## Description
Closes #6068

Sets the default background to dark until the application determines user preference.

## How Has This Been Tested?
Visually

## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

<!-- If you made any visual changes, provide screenshots of before/after, it its moving parts, please provide high quality gif, wemb or mp4 -->
![Screenshot 2023-08-12 215429](https://github.com/MudBlazor/MudBlazor/assets/57046415/472bc698-69c2-45a1-b8c5-cd40b568f8a8)

## Checklist:
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] The PR is submitted to the correct branch (`dev`).
- [x] My code follows the code style of this project.
- [ ] I've added relevant tests.

BieleckiLtd committed:
Adds dark background color on app load screen

mikes-gh commented:
But this will jar with light mode users.
Maybe check to see if there is a media query that can be used in modern browsers to detect dark mode.and keep the default as white if this function doesn't exist

ScarletKuro commented:
> But this will jar with light mode users

Yeah, I thought about it too. If person has light theme and even sets it explicitly on the website to have the light theme, during reloading page it will have dark theme, so the current solution is not ideal.

BieleckiLtd commented:
User-specific preferences cannot be retrieved and applied until the application has loaded. While media queries can theoretically apply system preferences, they cannot apply explicit user preferences. This is an area where I could enhance the solution.

As someone who uses both light and dark modes (depending on the time of day), I don't mind a brief dark loading screen when I want light mode, however, a bright loading screen at night is quite bothersome.

ScarletKuro commented:
Aren't the settings saved in the `userPreferences` in local storage? You can't execute C# while application is loading, but raw JS is available, you can check the local storage first if `DarkLightTheme` is `1` if the key doesn't exist, then default to check `window.matchMedia("(prefers-color-scheme: dark)")`

BieleckiLtd committed:
Add observer for user preferences or for the system settings

BieleckiLtd commented:
Summary:
1. Added a dark background using inline styling (opted against using a class file as it might still be loading).
2. Added JS to check a stored preference in the browser's localStorage to see if the user has specified a theme. If the user hasn't specified a preference, the code defaults to using the system's preference, determined by checking if the system prefers a light color scheme instead.
3. The code monitors for changes in the document's content. It detects the "loading screen" element being added to the DOM. If needed it adjusts its background color to light, then stops observing further DOM changes. 

On a slower machine a few milliseconds blip in light mode is unavoidable when the browser loads js, reads local storage, system preferences, and updates the DOM:

https://github.com/MudBlazor/MudBlazor/assets/57046415/1e646e9d-24b4-40c3-8622-5a6eab03c6b4

mckaragoz commented:
One more option is setting Mud primary color to the background

mikes-gh commented:
I tested the PR and it works for me 👍 
I did notice the dark light theme toggle switch does not detect the current theme on load but that is not this PR.

BieleckiLtd commented:
> One more option is setting Mud primary color to the background

Do you mean --mud-palette-primary? This is violet #594ae2 so not sure? I used values of --mud-palette-background which are either #ffffff or #1a1a27. Variables are not available until app loads so I used values directly.

henon commented:
Thanks @BieleckiLtd !
